### ISSUE TITLE
Enabling debug logging on gloo pods throws runtime error: invalid memory address or nil pointer

### ISSUE THREAD

**Issue by pszeto** [solo-io/gloo#6449]
### Gloo Edge Version

1.11.x (latest stable)

### Describe the bug

Unable to enable debugging on discovery service.

### Steps to reproduce the bug

1.   k port-forward -n gloo-system deployments/discovery 9091:9091
2. Navigate to http://localhost:9091
3. Change log level to debug and click.

```
2022/05/12 13:11:12 http: panic serving 127.0.0.1:56160: runtime error: invalid memory address or nil pointer dereference
goroutine 745 [running]:
net/http.(*conn).serve.func1(0xc001296c80)
	/usr/local/go/src/net/http/server.go:1824 +0x14c
panic(0x3d8d340, 0x5f05760)
	/usr/local/go/src/runtime/panic.go:971 +0x4c7
go.uber.org/atomic.(*Int32).Store(0x0, 0xc0ffffffff)
	/go/pkg/mod/go.uber.org/atomic@v1.9.0/int32.go:75 +0x5
go.uber.org/zap.AtomicLevel.SetLevel(0x0, 0xff)
	/go/pkg/mod/go.uber.org/zap@v1.19.1/level.go:102 +0x35
go.uber.org/zap.AtomicLevel.ServeHTTP(0x0, 0x474a570, 0xc0010861c0, 0xc0015e4500)
	/go/pkg/mod/go.uber.org/zap@v1.19.1/http_handler.go:90 +0x3a5
net/http.(*ServeMux).ServeHTTP(0xc000b56000, 0x474a570, 0xc0010861c0, 0xc0015e4500)
	/usr/local/go/src/net/http/server.go:2448 +0x1ab
net/http.serverHandler.ServeHTTP(0xc000e82000, 0x474a570, 0xc0010861c0, 0xc0015e4500)
	/usr/local/go/src/net/http/server.go:2887 +0x22b
net/http.(*conn).serve(0xc001296c80, 0x474f7b0, 0xc001e3f6c0)
	/usr/local/go/src/net/http/server.go:1952 +0x1c25
created by net/http.(*Server).Serve
	/usr/local/go/src/net/http/server.go:3013 +0x96e
```

Using curl to enable debugging
```
curl -X PUT -H "Content-Type: application/json" -d '{"level": "debug"}' http://localhost:9091/logging
```

Results in the same error.

Running : 
```
Server: {"type":"Gateway","enterprise":true,"kubernetes":{"containers":[{"Tag":"1.11.6","Name":"discovery-ee","Registry":"quay.io/solo-io"},{"Tag":"1.11.6","Name":"extauth-ee","Registry":"quay.io/solo-io"},{"Tag":"1.11.7","Name":"gateway","Registry":"quay.io/solo-io"},{"Tag":"1.11.6","Name":"gloo-ee-envoy-wrapper","Registry":"quay.io/solo-io"},{"Tag":"1.11.6","Name":"gloo-ee","Registry":"quay.io/solo-io"},{"Tag":"1.11.6","Name":"observability-ee","Registry":"quay.io/solo-io"},{"Tag":"1.11.6","Name":"rate-limit-ee","Registry":"quay.io/solo-io"},{"Tag":"6.2.4","Name":"redis","Registry":"docker.io"}],"namespace":"gloo-system"}}
```

### Expected Behavior

Debug logging should be enabled through the curl command.

**Comment by pszeto** [comment]
Work around:
Add the debug loglevel at the deployment `spec.template.spec.containers.env`
```        
- name: LOG_LEVEL
   value: debug
   ```

**Comment by nparry-zymergen** [comment]
Can confirm this is also the case on 1.9.16.

**Comment by nfuden** [comment]
https://github.com/solo-io/go-utils/pull/487 fixes this in 1.12 we still need to do appropriate backports

**Comment by pszeto** [comment]
Confirmed fixed in the latest 1.11.x stream (version 1.11.39)
